src/cirrus/lib2/events.py

```diff
@@ -0,0 +1,353 @@
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from functools import wraps
+from logging import Logger, getLogger
+from typing import Any, Callable, Dict, Optional
+
+import boto3
+
+from .enums import StateEnum, WFEventType
+from .eventdb import EventDB
+from .statedb import StateDB
+from .utils import PAYLOAD_ID_REGEX, SNSPublisher
+
+
+@dataclass
+class WorkflowEvent:
+    event_type: WFEventType
+    payload_id: str
+    isotimestamp: str
+    payload_url: Optional[str] = None
+    execution_arn: Optional[str] = None
+    error: Optional[Dict[str, Any]] = None
```

Comment:
Is this the right type? Isn't `error` just a `str`?

Reply:
Not the right type.  `Execution.error` in `update-state` uses a dict, but I see that it is distilled to a string before the `WorkflowEvent` is created.